test/stream.spec.ts

```diff
@@ -0,0 +1,66 @@
+
+import { expect } from 'aegir/chai'
+import * as lengthPrefixed from 'it-length-prefixed'
+import { pushable } from 'it-pushable'
+import { Message } from '../src/pb/message'
+import * as underTest from '../src/stream'
+
+const setup = (cb: { send: (bytes: Uint8Array) => void }): underTest.WebRTCStream => {
+  const datachannel = {
+    readyState: 'open',
+    send: cb.send,
+    close: () => {}
+  }
+  return new underTest.WebRTCStream({ channel: datachannel as RTCDataChannel, stat: underTest.defaultStat('outbound') })
+}
+
+const MAX_MESSAGE_SIZE = 16 * 1024
+
+describe('Max message size', () => {
+  it(`sends messages smaller or equal to ${MAX_MESSAGE_SIZE} bytes in one`, async () => {
```

Comment:
Thank you for adding tests.

Could you please add one more that tests that when the event timeout is reached an appropriate error is thrown and that the stream is closed correctly.

Reply:
Agreed, I added an e2e test here but it doesn't appear to work 😱 - https://github.com/libp2p/js-libp2p-webrtc/pull/165

Reply:
I did tests manually now, logging the configurations on both ends when runninng the browser-to-browser example and saw that the configurations are propagating through

Reply:
I also addded it to ```wrappedChannel``` now, I am not sure the implications of this, if it is only for the noise handshake?

Reply:
Would my manual test be fine for now before #165 is fixed? 

Perhaps we can add proper configuration testing later, also add bound checking/tests for configuration, so you don't create a configuration with unfeasible constraints

Reply:
I think it's fine - we can more comprehensive tests after #165